### PR TITLE
refactor(db)!: suppress Db events

### PR DESCRIPTION
**Pull request by emadum**
## Description

[NODE-1709](https://jira.mongodb.org/browse/NODE-1709) - Stop emitting topology events from `Db` 
[NODE-2474](https://jira.mongodb.org/browse/NODE-2474) - Fix double close event

**What changed?**

The `Db` class no longer inherits `EventEmitter`, and `MongoClient` no longer forwards topology events to `Db` objects. Some event handlers in the unit tests were updated to reflect this, and a deprecated `unifiedTopology: false` test was removed.

~~**Are there any files to ignore?**~~

### CODE REVIEW

File: test/functional/db.test.js [hunk 1]
```diff
@@ -183,7 +183,7 @@ describe('Db', function() {
           });
         };
 
-        automatic_connect_client.once('close', closeListener);
+        client.once('close', closeListener);
```
Comment (mbroadst) [inline test/functional/db.test.js]: `automatic_connect_client` seems particularly poorly named

File: lib/db.js [hunk 2]
```diff
@@ -140,7 +137,6 @@ const legalOptionNames = [
 function Db(databaseName, topology, options) {
```
[the remark this answers is not in the document]
Comment (emadum) [reply lib/db.js]: Good point, will do!

File: test/functional/db.test.js (hunk 1 above)
Comment (emadum) [reply test/functional/db.test.js]: Agreed, perhaps `auto_reconnect_client` would be better? There's also a mix of camel/snake case for variables being used in these tests, would be nice to make that more consistent eventually.

Comment (mbroadst) [reply test/functional/db.test.js]: Well I was actually thinking "this thing is a `Db` not a `MongoClient`!"

Comment (emadum) [reply test/functional/db.test.js]: Oh right, didn't read that one correctly. I'll clean that test up a bit.

File: lib/db.js [hunk 3]
```diff
@@ -791,7 +768,6 @@ Db.prototype.ensureIndex = deprecate(function(name, fieldOrSpec, options, callba
 
 Db.prototype.addChild = function(db) {
   if (this.s.parentDb) return this.s.parentDb.addChild(db);
-  this.s.children.push(db);
```
Comment (mbroadst) [inline lib/db.js]: I think when removing the internal `children` array, we should also just completely remove the concept of a `Db` having "children". I know it's a bit out of scope for the original ticket, but it seems like not much extra work.

Comment (emadum) [reply lib/db.js]: Done!

**Pull request merged**